hook.sh

```diff
@@ -279,18 +279,25 @@ while true; do
 
   display_warnings
 
-  # Ask the question (not using "read -p" as it uses stderr not stdout)
-  echo -en "${BLUE}Proceed with commit? [e/y/n/?] ${NC}"
+  if [ -t 1 ] || [ ! -z ${FAKE_TTY+x} ]; then
```

Comment:
This approach is working well. I don't love the test specific `FAKE_TTY` featuring in the hook itself, but I can't think of a way around this right now and I think the non-interactive support is worth it.

Personally, I'd prefer to exit with a guard statement rather than the `if`/`else`, e.g.

```sh
# if non-interactive don't prompt and exit with an error
if [ ! -t 1 ] && [ -z ${FAKE_TTY+x} ]; then
  exit 1
fi
```

But that's purely stylistic.

If you submit a separate MR with just the non-interactive fix I'll get this merged.

Thanks,
Tom